examples/compose/plot_compare_reduction.py

```diff
@@ -119,7 +117,7 @@
 grid.fit(digits.data, digits.target)
 
 # Delete the temporary cache before exiting
-rmtree(cachedir)
+memory.clear(warn=False)
```

Comment:
Have you checked that this also removes the location folder?

Reply:
yes, I've checked after calling 'clear()' location folder is removed. However, not main directory - 'cachedir' (in this case), but subdirectory - 'sklearn'.

Reply:
OK, then we still want to remove the `cachedir` though.

Reply:
I found smth that can meet our requirements: [clear_path](https://github.com/joblib/joblib/blob/master/joblib/_store_backends.py#L245). But, it requires one extra code line.